### PR TITLE
[fix](cloud-mow) Fix the issue of missing and removing some old version delete bitmap

### PR DESCRIPTION
**Pull request by hust-hhb**
### What problem does this PR solve?

Related PR: #40204

Problem Summary:
pr #40204 support removing old delete bitmap, however it donesn't consider boundary which will lead to miss some delete bitmap should be removed on be , it only affect local delete bitmap, the delete bitmap store in fdb is right, and these missing delete bitmaps can only be deleted by next base compaciton before this pr.

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

File: be/src/olap/tablet_meta.cpp [hunk 1]
```diff
@@ -1209,9 +1209,10 @@ void DeleteBitmap::remove_stale_delete_bitmap_from_queue(const std::vector<std::
                 }
                 auto start_bmk = std::get<1>(delete_bitmap_tuple);
                 auto end_bmk = std::get<2>(delete_bitmap_tuple);
+                //to remove delete bitmap doesn't contains end_bmk
```
Comment (zhannngchen) [inline be/src/olap/tablet_meta.cpp:1212]: Here to explain in detail, in the future, when others come to maintain the code here, may not be clear about the reason for the -1 operation here and ignore this detail.

File: regression-test/suites/compaction/test_cu_compaction_remove_old_version_delete_bitmap.groovy [hunk 2]
```diff
@@ -273,38 +301,42 @@ suite("test_cu_compaction_remove_old_version_delete_bitmap", "nonConcurrent") {
...
 
-            delete_bitmap_count = getDeleteBitmapStatus(backendId_to_backendIP[trigger_backend_id], backendId_to_backendHttpPort[trigger_backend_id], tablet_id).delete_bitmap_count
-            assertTrue(delete_bitmap_count == 6)
-            logger.info("delete_bitmap_count:" + delete_bitmap_count)
+            local_delete_bitmap_count = getLocalDeleteBitmapStatus(backendId_to_backendIP[trigger_backend_id], backendId_to_backendHttpPort[trigger_backend_id], tablet_id).delete_bitmap_count
+            ms_delete_bitmap_count = getMSDeleteBitmapStatus(backendId_to_backendIP[trigger_backend_id], backendId_to_backendHttpPort[trigger_backend_id], tablet_id).delete_bitmap_count
+            logger.info("local_delete_bitmap_count:" + local_delete_bitmap_count)
+            logger.info("ms_delete_bitmap_count:" + ms_delete_bitmap_count)
+            assertTrue(local_delete_bitmap_count == 6)
+            assertTrue(local_delete_bitmap_count == ms_delete_bitmap_count)
```
Comment (zhannngchen) [inline regression-test/suites/compaction/test_cu_compaction_remove_old_version_delete_bitmap.groovy:309]: check cardinality as well

**Comment by hust-hhb** [comment]
run buildall

**Comment by hust-hhb** [comment]
@zhannngchen already change this pr, take a look please

**Review by zhannngchen — APPROVED** [review]
LGTM

**Comment by hust-hhb** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**